Lib/idlelib/pyshell.py

```diff
@@ -1698,6 +1698,8 @@ def main():
         root.mainloop()
     root.destroy()
     capture_warnings(False)
+    import _tkinter
```

Comment:
What if add `del root` and two `gc.collect()`? Would it crash?

Reply:
Why would adding those cause a crash? Sorry, I don't understand the reasoning here, would you care to expand on this?

Reply:
This shouldn't cause a crash, but the check Serhiy suggested is to ensure that the interpreter isn't required for finalisation (i.e. if the root gets garbage collected, things are not going to go horribly wrong).

Reply:
AFAICT nothing else in our codebase calls `Tcl_Finalize`, `Tcl_FinalizeThread`, `Tcl_Exit` or anything else that triggers Tcl's finalization.

In addition, calling these multiple times shouldn't be an issue according to [the Tcl docs](https://www.tcl.tk/man/tcl8.6/TclLib/Exit.htm):

> `Tcl_Finalize` can be safely called more than once.